### PR TITLE
Rewrite planewave orbital set builder

### PR DESCRIPTION
**Pull request by ye-luo**
## Proposed changes
PWorbitalBuilder was based on WaveFunctionComponentBuilder
reworked PWorbitalSetBuilder is based on SPOSetBuilder and connected to the rest like other SPOSet builders.

## What type(s) of changes does this code introduce?
- Refactoring (no functional changes, no api changes)

### Does this introduce a breaking change?
- No

## What systems has this change been tested on?
laptop

## Checklist
- Yes. This PR is up to date with current the current state of 'develop'

### CODE REVIEW

**Comment by prckent** [comment]
Thanks for the update. I think this is very important to keep around as a check on the splines. It would make good batching+offloading exercise for someone as well..

File: src/QMCWaveFunctions/BsplineFactory/EinsplineSetBuilderESHDF.fft.cpp [hunk 1]
```diff
@@ -339,6 +368,127 @@ bool EinsplineSetBuilder::ReadOrbitalInfo_ESHDF(bool skipChecks)
...
+  if (root)
+  {
+    H5File.readEntry(MeshSize, "/electrons/psi_r_mesh");
+    H5File.readEntry(MeshSize, "/electrons/mesh");
+  }
+  myComm->bcast(MeshSize);
+  hasPsig = (MeshSize[0] == 0);
+  if (hasPsig)
+  {
+    int nallowed  = 257;
```
Comment (prckent) [inline src/QMCWaveFunctions/BsplineFactory/EinsplineSetBuilderESHDF.fft.cpp:386]: This seems to be new. Explanation required.

File: src/QMCWaveFunctions/BsplineFactory/EinsplineSetBuilder_createSPOs.cpp [hunk 2]
```diff
@@ -124,7 +124,6 @@ std::unique_ptr<SPOSet> EinsplineSetBuilder::createSPOSetFromXML(xmlNodePtr cur)
       "no"); // use old spline library for high-order derivatives, e.g. needed for backflow optimization
   std::string useGPU;
   std::string GPUsharing = "no";
-  std::string spo_object_name;
```
Comment (ye-luo) [inline src/QMCWaveFunctions/BsplineFactory/EinsplineSetBuilder_createSPOs.cpp:127]: This variable was never consumed and thus delete.

**Comment by ye-luo** [comment]
Test this please

**Comment by ye-luo** [comment]
@Hyeondeok-Shin this should unblock you from updating input files for runs using PW orbitals.

**Review by prckent — APPROVED** [review]

**Pull request merged**